Refactor references to submit support requests (#1467)
hashicorp/design-system: website/docs/about/contribution.md

While we don’t accept direct contributions of new components or patterns, we do welcome contributions in the following forms:

> * [Reporting](support#file-an-issue) or [fixing](#code-contributions) bugs and [giving feedback](https://docs.google.com/forms/d/e/1FAIpQLSdPShbSiUYgimppZeqWT99bVi6zLbKDThQUQTirH3RCj2okdg/viewform)

Reviewer: The "fixing" link doesn't really do anything because the page isn't long enough for the proper jump, so I've added an additional indicator of the section to refer to. I also found it weird that the "feedback" one was lumped together with the other two.

```suggestion
* [Reporting](support#file-an-issue) or fixing an issue or bug (see [Code contributions](#code-contributions))
* [Providing general feedback](https://docs.google.com/forms/d/e/1FAIpQLSdPShbSiUYgimppZeqWT99bVi6zLbKDThQUQTirH3RCj2okdg/viewform)
```

> * [Proposing new components or patterns](about/support#request-a-feature)

Reviewer: ```suggestion
* [Proposing new components or patterns](support#request-a-feature)
```

> * [Adding an icon](#icon-contributions)

Reviewer: I ran into the same issue with the skip link, where the page isn't long enough for the proper jump. 

```suggestion
* Adding an icon (see [Icon contributions](#icon-contributions))
```
